assembl/static2/js/app/components/form/datePickerFieldAdapter.jsx

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/*
+  DatePicker adapter for react-final-form that supports the following form:
+  { time: moment().utc() }
+
+  Errors are managed by a Mutator on the Form itself. As a result, to show errors
+  on DatePickerFieldAdapter, must add the `setFieldTouched` from final-form-set-field-touched
+  along with passing the form <FormApi> prop
+*/
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import * as React from 'react';
+import { type FieldRenderProps } from 'react-final-form';
+import { type FormApi } from 'final-form';
+import { ControlLabel, FormGroup } from 'react-bootstrap';
+
+import type { DatePickerType, DateTime, DatePickerValue } from './types.flow';
+import Error from './error';
+import { getValidationState } from './utils';
+
+type Props = {
+  editLocale: string,
+  picker?: DatePickerType,
+  placeHolder: string,
+  showTime: boolean,
+  hasConflictingDate: boolean,
+  input: {
+    name: string,
+    onChange: (SyntheticInputEvent<*> | any) => void,
+    value: DatePickerValue
+  },
+  onDateChange: ?(DateTime) => void,
+  form: FormApi,
+  children?: React.Node
+} & FieldRenderProps;
+
+const DatePickerFieldAdapter = ({
+  editLocale,
+  picker,
+  placeHolder,
+  showTime,
+  input: { name, value, onChange },
+  meta: { error, touched },
+  required,
+  hasConflictingDate,
+  onDateChange,
+  children,
+  form,
+  ...rest
+}: Props) => {
+  const onLocalizedChange = (e: DateTime): void => {
+    if (e !== value.time) {
+      if (form) {
```

Comment:
`if (e !== value.time && form) {`

Reply:
Ooops, this logic was flawed. Instead of putting form up there, instead, I shifted the logic below under the same conditional check as the form